libgenic/power.c

```diff
@@ -149,15 +206,25 @@ void parse_transfer(int i, double k, char * line, struct table *out_tab, int * I
      * */
     out_tab->logD[DELTA_BAR][i] = -1*transfers[1];
     out_tab->logD[DELTA_CDM][i] = -1*transfers[2];
-    /*This is ur if neutrinos are massless*/
-    out_tab->logD[DELTA_NU][i] = -1*transfers[3+nnu];
+    /*This should be the weighted average sum of the three neutrino species*/
+    const _omega_nu * Onu = &CP->ONu;
+    out_tab->logD[DELTA_NU][i] = 0;
+    for(j=0; j < nnu; j++)
+        out_tab->logD[DELTA_NU][i] = -1*transfers[4+j] * omega_nu_single(Onu, InitTime, j);
+    const double onu = get_omega_nu(&CP->ONu, InitTime);
+    /*Should be weighted bu omega_nu*/
+    out_tab->logD[DELTA_NU][i] /= onu;
     /*h_prime is entry 8 + nnu. t_b is 12 + nnu, t_ncdm[2] is 13 + nnu * 2.*/
     out_tab->logD[VEL_BAR][i] = transfers[12+nnu];
     out_tab->logD[VEL_CDM][i] = transfers[8+nnu] * 0.5;
-    out_tab->logD[VEL_NU][i] = transfers[13+nnu*2];
+    out_tab->logD[VEL_NU][i] = 0;
+    for(j=0; j < nnu; j++)
+        out_tab->logD[VEL_NU][i] = transfers[13 + nnu + j] * omega_nu_single(Onu, InitTime, j);
+    /*Should be weighted bu omega_nu*/
+    out_tab->logD[VEL_NU][i] /= onu;
 }
 
-void read_power_table(int ThisTask, const char * inputfile, const int ncols, struct table * out_tab, double scale, const int nnu, void (*parse_line)(int i, double k, char * line, struct table *, int *InputInLog10, const int nnu, double scale))
+void read_power_table(int ThisTask, const char * inputfile, const int ncols, struct table * out_tab, const double InitTime, void (*parse_line)(int i, double k, char * line, struct table *, int *InputInLog10, const double InitTime))
```

Comment:
What about taking out (*parse_line) with a typedef? This prototype is getting too long.

Reply:
Done